test/test-process.el

```diff
@@ -0,0 +1,21 @@
+;;; test-funcname-at-point.el --- Test the repl functions
+
+;;; Commentary:
+
+;; Test functions to interact with the REPL.
+
+;;; Code:
+
+(describe "`lua-start-process'"
+  (it "doesn't hang for an already-running process"
+    ;; Acquire a *lua* repl buffer
+    (lua-start-process)
+    ;; Input some text
+    (with-current-buffer lua-process-buffer
+      (insert "table.insert"))
+    ;; Try restarting the repl buffer
+    (lua-start-process)
+    ;; `lua-start-process' shouldn't hang
+    (expect t)))
```

Comment:
There's `test-inferior-process.el` already, so this should probably go there.

Aside from that, couple of minor improvements
- do `(call-interactively 'lua-start-process)`, to mimick `M-x lua-start-process` because there's actually a piece of code that behaves differently depending on whether or not it is an interactive call
- instead of `expect t` it could check that `lua-start-process` switches to repl buffer even if it already exists

Reply:
Done.

Reply:
Also, you should use the sharp-quote (#') syntax when referring to function symbols, because doing that will give you byte-compiler warnings if the function is not defined:
```emacs-lisp
(defun foo ())

(defun bar () (funcall 'fo) ; no warning
  )
(defun bar2 () (funcall #'fo) ; byte-compiler warning here
  )
```

Reply:
Great, thanks!